lib/datasource/npm/npmrc.ts

```diff
@@ -37,21 +34,56 @@ function envReplace(value: any, env = process.env): any {
   });
 }
 
-const envRe = regEx(/(\\*)\$\{([^}]+)\}/);
-// TODO: better add to host rules (#9588)
-function sanitize(key: string, val: string): void {
-  if (!val || envRe.test(val)) {
-    return;
+export function getMatchHostFromNpmrcHost(input: string): string {
+  if (input.startsWith('//')) {
+    const matchHost = input.replace('//', '');
+    if (matchHost.includes('/')) {
+      return 'https://' + matchHost;
+    }
+    return matchHost;
   }
-  if (key.endsWith('_authToken') || key.endsWith('_auth')) {
-    addSecretForSanitizing(val);
-  } else if (key.endsWith(':_password')) {
-    addSecretForSanitizing(val);
-    const password = fromBase64(val);
-    addSecretForSanitizing(password);
-    const username: string = npmrc[key.replace(':_password', ':username')];
-    addSecretForSanitizing(toBase64(`${username}:${password}`));
+  return input;
+}
+
+export function convertNpmrcToRules(npmrc: Record<string, any>): NpmrcRules {
+  const rules: NpmrcRules = {
+    hostRules: [],
+  };
+  const hostType = 'npm';
+  const hosts: Record<string, HostRule> = {};
+  for (const [key, value] of Object.entries(npmrc)) {
+    if (!is.string(value)) {
```

Comment:
```suggestion
    if (!is.nonEmptyString(value)) {
```

Hust to be save against empty strings?